File: Week 1/Problem Set 1/3. mario.c
#include <stdio.h>
#include <cs50.h>

void print_x_times(int x, char y)
{
    // print provided character provided number of times
    for (int i = 1; i <= x; i++)
    {
        printf("%c", y);
    }
}

int main(void)
{
    int num;
    do
    {
        num = get_int("Height: ");
    } while (num < 1 || num > 8);

    for (int i = 1; i <= num; i++)
    {
        print_x_times(num - i, ' ');
        print_x_times(i, '#');
        printf("  ");
        print_x_times(i, '#');
        printf("\n");
    }
}

File: Week 6/Problem Set 6/3. mario.py
def main():
    # ask for positive number (if user provides something else this will ask again)
    height = -1
    while(height < 1 or height > 8):
        try:
            height = int(input("Height (Please enter a positive number): "))
        except:
            continue

    # prints the pyramids
    for n in range(1, height + 1):
        print_x_times(height - n, " ")
        print_x_times(n, "#")
        print("  ", end="")
        print_x_times(n, "#")
        print("", end="\n")


def print_x_times(x, y):
    # take character and number x as a input and prints that character x times
    for n in range(x):
        print(y, end="")


main()